### PR TITLE
OEL-579 - Whitelabel - Medley branch with BCL 0.9.0 upgradel

### PR DESCRIPTION
**Pull request by drishu**
Jira:
- https://citnet.tech.ec.europa.eu/CITnet/jira/browse/OEL-579
Contains:
- Header: https://citnet.tech.ec.europa.eu/CITnet/jira/browse/OEL-587
- Footer: https://citnet.tech.ec.europa.eu/CITnet/jira/browse/OEL-589
- Language switcher Modal: https://citnet.tech.ec.europa.eu/CITnet/jira/browse/OEL-648
- Link corporate logos: https://citnet.tech.ec.europa.eu/CITnet/jira/browse/OEL-547

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
         "drupal/core": "^8.9 || ^9.1",
         "drupal/ui_patterns": "^1.0",
         "drupal/ui_patterns_settings": "^1.0",
-        "openeuropa/oe_bootstrap_theme": "0.181020211745"
+        "openeuropa/oe_bootstrap_theme": "dev-OEL-579-all"
```
Comment (drishu) [inline composer.json]: I prepared a new release 0.211020211256 from the 579-all branch, we can use that (you can always request one when needed, so you that one and revert the other composer changes.

File: modules/oe_whitelabel_helper/src/Plugin/Block/CorporateEuLogoBlock.php [hunk 2]
```diff
@@ -83,19 +84,21 @@ public function build(): array {
     $image = [
       '#theme' => 'image',
       '#uri' => $logo_path . '/logo-eu--' . $language . '.svg',
-      '#width' => '240px',
-      '#height' => '60px',
       '#alt' => $title,
       '#title' => $title,
     ];
-    $mobile_url = file_create_url($logo_path . '/mobile/logo-eu--' . $language . '.svg');
```
Comment (drishu) [inline modules/oe_whitelabel_helper/src/Plugin/Block/CorporateEuLogoBlock.php:91]: removing this is not good, see https://oelibrary.netlify.app/default/?path=/story/compositions-header--header&args=light:false, the EU logo has a mobile variant, that is what the picture/source setup is for, that is how browsers know to switch the image when in mobile

File: modules/oe_whitelabel_helper/tests/src/Kernel/CorporateLogoBlockTest.php [hunk 3]
```diff
@@ -121,11 +121,8 @@ public function testLogoBlockRendering(string $lang): void {
     $this->assertCount(1, $logo);
     $expected = "/themes/custom/oe_whitelabel/modules/oe_whitelabel_helper/images/logos/eu/logo-eu--{$lang}.svg";
     $this->assertSame($expected, $logo->attr('src'));
-    $picture = $actual->filter('picture');
```
Comment (drishu) [inline modules/oe_whitelabel_helper/tests/src/Kernel/CorporateLogoBlockTest.php:124]: put this back one you put the mobile setup back

File: phpunit.xml.dist [hunk 4]
```diff
@@ -11,7 +11,6 @@
   <testsuites>
     <testsuite name="OpenEuropa Whitelabel Theme">
       <directory>./tests/</directory>
-      <directory>./modules/**/tests/</directory>
```
Comment (drishu) [inline phpunit.xml.dist]: we don't test submodules anymore ?

File: templates/layouts/ec_footer_bottom_left.html.twig [hunk 5]
```diff
@@ -1 +1 @@
-<h4>{{ pattern('link', title.link) }}</h4>
+<p class="fw-bold pb-2 mb-2"> European Commission </p>
```
Comment (drishu) [inline templates/layouts/ec_footer_bottom_left.html.twig]: this needs to be translated |t

File: templates/layouts/ec_footer_middle_row.html.twig [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<p class="fw-bold pb-2 mb-2">More information on:</p>
```
Comment (drishu) [inline templates/layouts/ec_footer_middle_row.html.twig]: translate static strings, also, we need to import the translations of these static strings, see: 

- https://github.com/openeuropa/oe_theme/tree/3.x/modules/oe_theme_helper/translations
- https://github.com/openeuropa/oe_theme/blob/9ae0e316b23bdb4c95f749145f04906e48ae4e31/modules/oe_theme_helper/oe_theme_helper.info.yml#L16
- https://github.com/openeuropa/oe_theme/blob/9ae0e316b23bdb4c95f749145f04906e48ae4e31/modules/oe_theme_helper/oe_theme_helper.module#L221

File: templates/layouts/eu_footer_middle.html.twig [hunk 7]
```diff
@@ -1,4 +1,4 @@
-<h4> {{ contact_eu_title|t }}  </h4>
+<p class="fw-bold border-bottom pb-2 mb-2"> {{ contact_eu_title|t }}  </p>
```
Comment (drishu) [inline templates/layouts/eu_footer_middle.html.twig]: Please make a test, isn't contact_eu_title already sent translated (if the translation is added ?), same for below ones

File: templates/overrides/navigation/block--system-branding-block.html.twig [hunk 8]
```diff
@@ -16,14 +16,10 @@
 {% block content %}
   <div class="d-flex align-items-center my-3">
   {% if site_logo %}
-    <a href="{{ path('<front>') }}" rel="home" class="site-logo">
```
Comment (drishu) [inline templates/overrides/navigation/block--system-branding-block.html.twig:19]: the fact that it's not linked is a mistake on the FRONT side, please put back and create a ticket for them

File: templates/overrides/navigation/oe-corporate-blocks-eu-footer.html.twig [hunk 9]
```diff
@@ -54,14 +54,14 @@
 
   {# Follow section must always be the last on the left. #}
   {% set _section_2 = _section_2|merge([{
-    'title': 'Follow us'|t,
+    'title': 'Follow us on social media'|t,
```
Comment (drishu) [inline templates/overrides/navigation/oe-corporate-blocks-eu-footer.html.twig:57]: check which strings we have translations for in oe_theme to copy over, and report the ones we don't, so we can ask for them

File: templates/overrides/page/page.html.twig [hunk 10]
```diff
@@ -96,7 +97,11 @@
   </main>
 
   {% if page.footer %}
-    <footer class="page__footer">
+    {% if "ecfooterblock" in page.footer|keys %}
```
Comment (drishu) [inline templates/overrides/page/page.html.twig]: I don't like the if, looks bad, I also see that the header blue bar color is different between EU and EC in netlify, so, we can go ahead and create a configuration for the theme here admin/appearance/settings/oe_whitelabel, where we set if it's a EU ot EC website. With that config variable (and we can assume a default) we can decide in situations like the header and the footer if it should be one or the other. See https://github.com/openeuropa/oe_theme/blob/3.x/oe_theme.theme#L44

File: templates/patterns/header_nav/pattern-header_nav.html.twig [hunk 11]
```diff
@@ -1,38 +1,35 @@
-<header>
-  <div class="container">
-    <div class="oe-header">
+{% if 'corporateeclogoblock' in navbar_branding|keys %}
```
Comment (drishu) [inline templates/patterns/header_nav/pattern-header_nav.html.twig]: again, decide via component_library

File: templates/patterns/modal/modal-language.html.twig [hunk 12]
```diff
@@ -12,7 +12,8 @@
     - footer
       - label
 #}
-
+{% set eu_official_languages = 'EU official languages' %}
```
Comment (drishu) [inline templates/patterns/modal/modal-language.html.twig]: |t for both, see if we can get the translations from oe_theme or we need to ask for them

File: tests/src/Kernel/FooterBlockTest.php [hunk 13]
```diff
@@ -81,7 +81,7 @@ public function testEcFooterBlockRendering(): void {
     $actual = $crawler->filter('#block-ecfooterblock');
     $this->assertCount(1, $actual);
     $sections = $actual->filter('.row');
-    $this->assertCount(2, $sections);
+    $this->assertCount(3, $sections);
```
Comment (drishu) [inline tests/src/Kernel/FooterBlockTest.php]: we added a lot of new markup, strings and css classes, lets test for these

File: tests/src/Kernel/SiteBrandingBlockTest.php [hunk 14]
```diff
@@ -75,8 +75,6 @@ public function testBlockRendering(): void {
 
     $actual = $crawler->filter('.site-name');
     $this->assertCount(1, $actual);
-    $link = $actual->filter('.h1.text-white.text-decoration-none.align-bottom');
```
Comment (drishu) [inline tests/src/Kernel/SiteBrandingBlockTest.php:78]: put this back once you put back the link, and I think we have new css classes to check against

**Review by abel-santos-corral — APPROVED** [review]

**Review by escuriola — APPROVED** [review]

**Pull request merged**